src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
                 {
-                    if (e.NewValue is ITrackCollectionViewModel zt)
-                    {
-                        zt.Tracks.CollectionChanged += trackCollection.Tracks_CollectionChanged;
-                    }
-
-                    if (e.OldValue is ITrackCollectionViewModel zte)
+                    var newItem = new ZuneTrackCollectionItem
                     {
-                        zte.Tracks.CollectionChanged -= trackCollection.Tracks_CollectionChanged;
-                    }
-                }
-            }));
+                        Track = (TrackViewModel)x,
+                        ParentCollection = Collection
+                    };
 
-        /// <summary>
-        /// Backing dependency property for <see cref="AlbumArtistCollection"/>.
-        /// </summary>
-        public IAlbumCollectionViewModel? AlbumArtistCollection
-        {
-            get { return (IAlbumCollectionViewModel)GetValue(AlbumArtistCollectionProperty); }
-            set { SetValue(AlbumArtistCollectionProperty, value); }
-        }
-
-        /// <summary>
-        /// Dependency property for <ses cref="IAlbumCollectionViewModel" />.
-        /// </summary>
-        public static readonly DependencyProperty AlbumArtistCollectionProperty =
-            DependencyProperty.Register(nameof(AlbumArtistCollection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, null));
-
-        private async void Tracks_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
-        {
-            if (PART_ArtistColumn == null)
-                return;
+                    AttachEvents(newItem);
+                    return newItem;
+                }));
+            }
 
-            if (Collection is AlbumViewModel)
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Remove)
             {
-                foreach (var track in Collection.Tracks)
+                foreach (var item in e.OldItems)
                 {
-                    var artists = await track.GetArtistItemsAsync(track.TotalArtistItemsCount, 0).ToListAsync();
-
-                    if (artists.Count > 1)
+                    var target = TrackItems.FirstOrDefault(x => x.Track == item);
+                    if (target is not null)
                     {
-                        PART_ArtistColumn.Visibility = Visibility.Visible;
-                        return;
+                        _trackItems.Remove(target);
+                        DetachEvents(target);
                     }
                 }
-                PART_ArtistColumn.Visibility = Visibility.Collapsed;
             }
-            else
+
+            if (e.Action == System.Collections.Specialized.NotifyCollectionChangedAction.Move)
             {
-                PART_ArtistColumn.Visibility = Visibility.Collapsed;
+                for (var i = 0; i < e.OldItems.Count; i++)
```

Comment:
Didn't know you can move tracks?

Reply:
I'm not sure what you mean by this. We're handling `NotifyCollectionChangedAction.Move`.

Reply:
Yes this usually occurs when someone drag and drop an item, remember share where you could drag and drop favorited items, well that was more of a TwoWay binding example.

Reply:
Well, not necessarily, It just seemed extra handling.